hello-world.md

```diff
@@ -1,15 +1,30 @@
 # Hello, World!
 
-["Hello, World!"](http://en.wikipedia.org/wiki/%22Hello,_world!%22_program) is typically the first 
-program that anyone writes in any language. It is, by design, an example of the absolute simplest 
-program that can possibly be written in any given language. Whenever an experienced programmer 
-starts to learn a new programming language, it is very common that the first program they will 
-write is some form of "Hello, World!" This helps them do two things:
-
-1. Verify that their developement environment is working properly
-2. Prove to themselves that they can write a very simple program
-
-Basically, a "Hello, World!" program simply outputs the text "Hello, World!" to the screen 
-or console. (NOTE: if you don't know what a console aka command-line interface aka command 
-window aka terminal window is, you should take a few minutes to 
-[read the Wikipedia article about the Command-line interface](http://en.wikipedia.org/wiki/Command-line_interface).)
\ No newline at end of file
+Every track at Exercism starts with ["Hello, World!"](http://en.wikipedia.org/wiki/%22Hello,_world!%22_program). Hello World is the traditional first program for beginning programmers. As programmers mature, they eventually want to test their code.
+
+Here at Exercism we encourage [Test Driven Development](http://en.wikipedia.org/wiki/Test-driven_development) (TDD). Test Driven Development is where you write your tests before writing any functionality. This allows you to consider the requirements of the code, and any edge cases that you may need to support.
+
+Once you finish your tests, you write the functionality. When all the functionality is finished, you can write tests for your next feature. One of the best features of tests is that you can very easily check for regression.
+
+We encourage users to use TDD when solving problems to help form good coding habits,
+and help simulate working on **Real World™** projects.
+
+## A typical TDD workflow on Exercism:
+
+1. Run the test file and pick one test that's failing.
+2. Write some code to fix the test you picked.
+3. Re-run the tests to confirm the test is now passing.
+4. Repeat from step 1.
+5. [Submit your solution](http://help.exercism.io/submitting-exercises.html).
+
+## Instructions
+
+It's important to remember that the goal is to make code as expressive and readable as we can. Therefore, code submitted should be as general as possible and not just try to pass the tests, but also work for any input that matches the specification.
```

Comment:
This is actually a fine line... we don't want to encourage speculative generality / premature generalization etc, but we also don't want people to submit a solution to `bob` that is a massive case statement for all the explicit string inputs in the tests.

It's kind of like _general_ but not _flexible_.